### PR TITLE
Adding core meeting info

### PR DESCRIPTION
**Pull request by jasminejaksic**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: GROUPS.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ you'll have access to all the docs and won't need to request access for each of
 
 | Name | Leads | Mailing List | Example Topics | Meetings |
 |------|-------|---------|----------|----------|
-| Core | Sven Mawson (Google), Louis Ryan (Google), Martin Taillefer (Google), Shriram Rajagopalan (IBM), Dan Berg (IBM), Lin Sun(IBM) | [istio-core@](https://groups.google.com/forum/#!forum/istio-core) | Configuration, Performance, Stability | TBD
+| Core | Sven Mawson (Google), Louis Ryan (Google), Martin Taillefer (Google), Shriram Rajagopalan (IBM), Dan Berg (IBM), Lin Sun(IBM) | [istio-core@](https://groups.google.com/forum/#!forum/istio-core) | Configuration, Performance, Stability | [Mondays at 13:00 PT (weekly)](https://groups.google.com/forum/#!forum/istio-0-3-progr)
```
Comment (geeknoid) [inline GROUPS.md]: I don't think that's the right meeting link :-)

Should we add the config working group info? Sven never got around to doing that.

Comment (jasminejaksic) [reply GROUPS.md]: That was very sloppy of me - fixed the incorrect link. 
About config, Joy is polling the date/time/frequency for meetings, we could wait until that settles before adding a row - wdyt?

**Comment by hklai** [comment]
Perhaps we should rename the meeting as "istio-0-3-progr" looks specific to 0.3 only.
Thought?

**Comment by jasminejaksic** [comment]
Andy, changed it to istio-core (please update the meeting id to reflect the change).

**Pull request closed without merging**